CountedSet.playground/Contents.swift

```diff
@@ -0,0 +1,58 @@
+import UIKit
+
+
+struct CountedSet<Element: Hashable>: ExpressibleByArrayLiteral {
+    
+    
+    var dict: [Element: Int] = [:]
+    
+    
+    init(arrayLiteral elements: Element...) {
+        for element in elements {
+            insert(element)
+        }
+    }
+    
+    mutating func insert(_ element: Element) {
+        dict[element] = (dict[element] ?? 0) + 1
+    }
+    
+    mutating func remove(_ element: Element) -> Int{
+        var count = dict[element] ?? 0
+        
+        if count > 0 {
+            dict[element] = count - 1
+            count -= 1
+        }
+        return count
+    }
+    
+    func count() -> Int {
+        guard !dict.isEmpty else { return 0 }
```

Comment:
I like that you were thinking of edge cases, however, I don't think this is necessary as if the dict is empty, the count will indeed be 0.